pkg/cloudprovider/provider/anexia/network_provisioning.go

```diff
@@ -0,0 +1,122 @@
+package anexia
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/kubermatic/machine-controller/pkg/apis/cluster/common"
+	anxtypes "github.com/kubermatic/machine-controller/pkg/cloudprovider/provider/anexia/types"
+	anxclient "go.anx.io/go-anxcloud/pkg/client"
+	anxaddr "go.anx.io/go-anxcloud/pkg/ipam/address"
+	anxvm "go.anx.io/go-anxcloud/pkg/vsphere/provisioning/vm"
+	"go.uber.org/zap"
+)
+
+func networkInterfacesForProvisioning(ctx context.Context, log *zap.SugaredLogger, client anxclient.Client) ([]anxvm.Network, error) {
+	reconcileContext := getReconcileContext(ctx)
+
+	config := reconcileContext.Config
+	status := reconcileContext.Status
+
+	if status.Networks == nil || len(status.Networks) != len(config.Networks) {
+		if len(status.Networks) != 0 && len(status.Networks) != len(config.Networks) {
+			log.Warn("size of status.Networks != config.Networks, this should not happen in normal operation - ignoring existing status")
+		}
+
+		status.Networks = make([]anxtypes.NetworkStatus, len(config.Networks))
+	}
+
+	ret := make([]anxvm.Network, len(config.Networks))
+	for netIndex, network := range config.Networks {
+		networkStatus := &status.Networks[netIndex]
+		addresses := make([]string, len(network.Prefixes))
+
+		for prefixIndex, prefix := range network.Prefixes {
+			if networkStatus.Addresses == nil {
+				if len(networkStatus.Addresses) != 0 && len(networkStatus.Addresses) != len(network.Prefixes) {
+					log.Warnf("size of status.Networks[%v].Addresses != config.Networks[%v].Prefixes, this should not happen in normal operation - ignoring existing status", netIndex, netIndex)
+				}
+
+				networkStatus.Addresses = make([]anxtypes.NetworkAddressStatus, len(network.Prefixes))
+			}
+
+			reservedIP, err := getIPAddress(ctx, log, &network, prefix, &networkStatus.Addresses[prefixIndex], client)
+			if err != nil {
+				return nil, newError(common.CreateMachineError, "failed to reserve IP: %v", err)
+			}
+
+			addresses[prefixIndex] = reservedIP
+		}
+
+		ret[netIndex] = anxvm.Network{
+			VLAN:    network.VlanID,
+			IPs:     addresses,
+			NICType: anxtypes.VmxNet3NIC,
+		}
+	}
+
+	return ret, nil
+}
+
+var _engsup3404mutex sync.Mutex
```

Comment:
Same goes for this mutex. A little comment that it's used to prevent VM race conditions doesn't hurt IMO. (and yes, I know that it's a leftover from the previous code. let's fix it quickly when we touch it anyway. ^^)

Reply:
The ticket even got marked as completed, maybe we can remove the mutex completely?

Reply:
The ticket got marked as completed, but if you read the last comment, it just acknowledges this still happening if two calls happen "nearly on the same millisecond" - I'm not happy with this, will add a comment though (but I just moved this code around ^^')